src/main/java/frc/robot/util/Logger.java

```diff
@@ -0,0 +1,153 @@
+package frc.robot.util;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.StringJoiner;
+import java.util.function.Supplier;
+
+/**
+ * Responsible for recording the sensor data
+ */
+public class Logger {
+
+    private String path;
+    private File file;
+
+    List<DynamicField> dynamicFields;
+
+    // TODO: Look into the datalog tool to be able to automatically pull the log file off the robot
+    // TODO: Use the RobotSimulation to test the logging utility
+    // TODO: Track Autonomous and Teleop Status
+    public Logger(String p, String name) {
+        path = p;
+        try {
+            file = new File(path + name + ".csv");
+            if (file.createNewFile()) {
+                System.out.println("File Created: " + file.getName());
+            } else {
+                System.out.println("File already exists.");
+            }
+
+            // Wipes the log if it already contains text
+            FileWriter myWriter = new FileWriter(file,false);
+            myWriter.close();
+
+        } catch (IOException e) {
+            System.out.println("Unable to create a file. ");
+            e.printStackTrace();
+        }
+
+        dynamicFields = new ArrayList<>();
+    }
+
+    public void log() {
+        try {
+            // TODO: Do not close the writer after each write
+            FileWriter myWriter = new FileWriter(file,true);
```

Comment:
I think you can accomplish this by simply initializing a private instance variable in your constructor.